src/test/java/org/broadinstitute/hellbender/utils/read/GATKReadAdaptersUnitTest.java

```diff
@@ -1378,5 +1378,95 @@ public void testUnmappedUnpairedDistinctionmates(final GATKRead read, final bool
+    public void testGetPairOrientationFailures(final GATKRead read) {
+        final SamPairUtil.PairOrientation pairOrientation = read.getPairOrientation();
+    }
+
+    @DataProvider(name = "pair_orientation")
+    public Object[][] pairOrientation() {
+        final GATKRead forwardStrandInnie = ArtificialReadUtils.createArtificialRead("100M");
+        forwardStrandInnie.setIsPaired(true);
+        forwardStrandInnie.setIsReverseStrand(false);
+        forwardStrandInnie.setMateIsReverseStrand(true);
+        forwardStrandInnie.setPosition(new SimpleInterval("1", 100, 200));
+        forwardStrandInnie.setMatePosition(new SimpleInterval("1", 200, 300));
+        forwardStrandInnie.setFragmentLength(200);
+
+        final GATKRead reverseStrandInnie = ArtificialReadUtils.createArtificialRead("100M");
+        reverseStrandInnie.setIsPaired(true);
+        reverseStrandInnie.setIsReverseStrand(true);
+        reverseStrandInnie.setMateIsReverseStrand(false);
+        reverseStrandInnie.setPosition(new SimpleInterval("1", 200, 300));
+        reverseStrandInnie.setMatePosition(new SimpleInterval("1", 100, 200));
+        reverseStrandInnie.setFragmentLength(200);
+
+        final GATKRead forwardStrandOutie = ArtificialReadUtils.createArtificialRead("100M");
+        forwardStrandOutie.setIsPaired(true);
+        forwardStrandOutie.setIsReverseStrand(false);
+        forwardStrandOutie.setMateIsReverseStrand(true);
+        forwardStrandOutie.setPosition(new SimpleInterval("1", 200, 300));
+        forwardStrandOutie.setMatePosition(new SimpleInterval("1", 100, 200));
+        forwardStrandOutie.setFragmentLength(-200);
+
+        final GATKRead reverseStrandOutie = ArtificialReadUtils.createArtificialRead("100M");
+        reverseStrandOutie.setIsPaired(true);
+        reverseStrandOutie.setIsReverseStrand(true);
+        reverseStrandOutie.setMateIsReverseStrand(false);
+        reverseStrandOutie.setPosition(new SimpleInterval("1", 100, 200));
+        reverseStrandOutie.setMatePosition(new SimpleInterval("1", 200, 300));
+        reverseStrandOutie.setFragmentLength(-200);
+
+        final GATKRead forwardStrandTandem = ArtificialReadUtils.createArtificialRead("100M");
+        forwardStrandTandem.setIsPaired(true);
+        forwardStrandTandem.setIsReverseStrand(false);
+        forwardStrandTandem.setMateIsReverseStrand(false);
+        forwardStrandTandem.setPosition(new SimpleInterval("1", 100, 200));
+        forwardStrandTandem.setMatePosition(new SimpleInterval("1", 200, 300));
+
+        final GATKRead reverseStrandTandem = ArtificialReadUtils.createArtificialRead("100M");
+        reverseStrandTandem.setIsPaired(true);
+        reverseStrandTandem.setIsReverseStrand(true);
+        reverseStrandTandem.setMateIsReverseStrand(true);
+        reverseStrandTandem.setPosition(new SimpleInterval("1", 100, 200));
+        reverseStrandTandem.setMatePosition(new SimpleInterval("1", 200, 300));
+
```

Comment:
May I also suggest adding test cases:
* mates mapped to different chromosomes
Of course if you believe getting pair orientation from such mate pairs to be non-sense, then it should be move to the test cases expecting exceptions.

Reply:
@SHuang-Broad What's your opinion?  It doesn't seem like any of the existing values of `SamPairUtil.PairOrientation` make sense.  If it weren't in htsjdk I would be tempted to add `DIFFERENT_CHROMOSOMES` to the enum.  I could imagine throwing an error and leaving it to the calling code to check.  I could also imagine having this method return an `Optional<SamPairUtil.PairOrientation>` and reserving an empty result for this case.